src/frontends/pytorch/src/op/gcd.cpp

```diff
@@ -0,0 +1,62 @@
+#include <ngraph/function.hpp>
+#include <openvino/openvino.hpp>
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/loop.hpp"
+#include "openvino/op/mod.hpp"
+#include "openvino/op/not_equal.hpp"
+#include "openvino/op/parameter.hpp"
+#include "openvino/op/reduce_logical_or.hpp"
+#include "openvino/op/select.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_gcd(const NodeContext& context) {
+    num_inputs_check(context, 2, 2);
+    auto x = context.get_input(0);
+    auto y = context.get_input(1);
+
+    auto zero = context.mark_node(v0::Constant::create(element::i32, Shape{}, {0}));
```

Comment:
`x` and/or `y` can have `int64` type. Please use `ConvertLike` operation for `zero` and `align_eltwise_input_types` helper function for `x` and `y` to support such case. Also please add a test with mixed `int32` and `int64` types.